src/d3a/d3a_core/redis_connections/redis_communication.py

```diff
@@ -41,20 +43,26 @@
 ZIP_RESULTS_CHANNEL = "d3a-zip-results"
 ZIP_RESULTS_KEY = "d3a-zip-results-key/"
 
+if TYPE_CHECKING:
+    from d3a.models.area import Area
+
 
 class RedisSimulationCommunication:
     def __init__(self, simulation, simulation_id, live_events):
-        if simulation_id is None:
```

Comment:
Can you explain a bit why this is needed to be removed ? I am asking because if the simulation_id is None then there is indeed no need to instantiate all members of the constructor.

Reply:
If `simulation_id  is None` I am indeed not instantiating all elements of the constructor, give ~5 lines below `if simulation_id is not None:`  a look

Reply:
I think Spyros meant that the original behavior (returning None if the simulation id is None) makes more sense
If the simulation id is None, imo we should raise an exception here

Reply:
Before `RedisSimulationCommunication` was returning None for the case of local run when there is no need of `stop/pause/resume/live-event & bulk-live-event` . Now the same `RedisSimulationCommunication` is used for local run too via adding a new callback to channel `area-map` for local run. Thats why I have now skipped an early return of this class constructor.

Reply:
How is it possible that the `simulation_id` is None and still you're getting the value of `d3a.constants.CONFIGURATION_ID`, isn't it also None in this case ?

Reply:
`d3a.constants.CONFIGURATION_ID` is an empty string for local run.

Reply:
IMO we should always rely on this `d3a.constants.CONFIGURATION_ID` variable then, and skip passing the `simulation_id`
And we should also remove the `if simulation_id is not None` check, because it doesn't make sense to only have an `area_mapping_callback` in case there is no simulation_id, if we're to interact with the simulation anw let's have all endpoints working (stop, pause, resume ...)

Reply:
`simulation_id ` being part of `RedisSimulationCommunication`being there earlier, but I agree and would incline to remove it now

Reply:
This approach is not going to work, unless we also modify the d3a-web (which I would not agree). simulation_id is the redis job id that is different for each simulation run, even for the same configuration. configuration_id is assigned whenever a configuration/collaboration/CN is created, and is constant no matter how many times one relaunches the same configuration. Therefore the simulation_id in the Redis channels names is really needed and will lead to a bug if configuration_id is used instead (the simulation won't be able to be stopped from the UI). 

Please be careful when adapting Redis channels in the future, because there are multiple services that might rely on them and one does not have the liberty to change at will without reviewing other services !